### ISSUE TITLE
Replace GroupID & marker component with in-editor functionality

### ISSUE THREAD

**Issue by Wolfenswan** [ferstaberinde/F3#715]
Groups now have their own init & setting fields.

The callsign can now be set directly in the editor.
The group marker function could be called directly from each groups init field. This requires some tests to check if it works nicely with JIPs and MP.

**Comment by Wolfenswan** [comment]
I've begun working on this and I think I quite like to go ahead with it.

What I imagine as implementation:
* Group IDs/Callsigns can be set directly in a dedicated field in the editor. Both sqfs for the component would no longer be necessary.
* Group Markers would be called from the group's init field with a line such as:
```nul = [this, "b_hq","ColorRed","HQ"] spawn f_fnc_localGroupMarker;```
Directly referring to the marker class makes it easier to customize the component and use all possible marker types from cfgMarkers (https://community.bistudio.com/wiki/cfgMarkers)
* By default this would only show the group to players of the same faction. To enable display to other factions/sides a fifth optional parameter would be available:
```nul = [this, "b_hq","ColorRed","HQ",[independent,"blu_t_f"]] spawn f_fnc_localGroupMarker;```
In this case the marker would also be shown to players on the indy side and of the nato pacific faction
* Specialist markers (UAV & medic) would be called directly from the unit init, with the same parameters as the groupMarker function

**Comment by darkChozo** [comment]
You could include a killswitch, something like if f_var_groupMarker == false then you just disable the whole component.

It would be nice if the script is structured in such a way that it's easy to override the faction parameter. Ie. an easy way to say that blu_g_f and blu_t_f can see blu_f markers without having to go over each individual blu_f unit.

**Comment by Wolfenswan** [comment]
Good point, would probably be more convenient to add a global variable to the init.sqf to determine who can see which marker.

That said, it wouldn't need to be every blu_f unit but every blu_f group; so maybe... 15 or so tops.

**Comment by ferstaberinde** [comment]
This sounds great. A couple of requests (some of which echo posts above):

1. Can we have 'component on/off' line in init.sqf

2. To determine who can see what, could we simply have arrays in init.sqf, such as:

markersSeenBy_FactionName1 = [FactionName1,FactionName2,FactionName3];
markersSeenBy_FactionName4 = [FactionName4,FactionName5,FactionName6];

**Comment by Wolfenswan** [comment]
>      markersSeenBy_FactionName1 = [FactionName1,FactionName2,FactionName3];
markersSeenBy_FactionName4 = [FactionName4,FactionName5,FactionName6];

This method would reduce functionality. Currently you explicitly show only certain *groups* to other factions (by adding the line to the relevant faction scope in the group makers script). The blanket approach would remove that functionality.

One possibility would be an array of stacked arrays:
```f_var_showMarkers = [["Blu_T_f",GrpNATO_DC]];```
But that's a mess.

I think the initially suggested method, by passing it as a parameter directly in the group init is preferable. How often do you need to show an entire ORBAT to a different faction?

**Comment by Wolfenswan** [comment]
There is a third, perhaps better suited solution (at least for the 3-4-0 release) as it's a middle ground.

Currently the mapMarker component uses the group variable, which is set in the unit init of each unit belonging to the group (```GrpAAF_CO = group this;```). The group ID can now be set directly in the group's config field.

This means we can remove the ```group this``` from all unit inits and use the identifier in the group window while keeping the current map Marker component.

Using this solution we could later introduce the method suggested above at any point with no need to undo any work done for e.g. new ORBATs.

The downside is that we'd keep the slightly bloated groupMarker.sqf. The bloat here could be countered by splitting it into subfiles by faction (or side?) and using #include, similar to how it's done in assignGear.

**Comment by ferstaberinde** [comment]
@Wolfenswan - I'm merging in PR #724 but to be clear, before we can close THIS issue the following must happen, right?

ToDo (in a future PR): Go through all groups and update them to the new standard. This requires:
Setting group variable & callsign in the group configuration window
Removing GrpX_Y = group this from all unit inits.

**Comment by Wolfenswan** [comment]
Exactly.
The work process in detail:
* Pick a group, note down it's variable and ID. (e.g. GrpAAF_DC & AAF DC). Group IDs are found here: http://ferstaberinde.com/f3/en//index.php?title=F3_Folk_ARPS_Group_IDs the group variable can be obtained by checking the init field of one of the group's units.
* Open the init field for each unit and delete the '''GrpXXX_YY = group this;''' line. Only the assignGear function should left.
* Open the init field for the group (double clicking the group symbol next to/above the units). Add the group variable and the group ID to the top section.
* Rinse & repeat.
* Test by confirming that the following are working: Group Markers & ORBAT notes (the orbat notes need to display the group ID correctly)

**Comment by ferstaberinde** [comment]
To match the old syntax, when we make these edits callsigns should be "NATO A1 - " (note the ' - ' after the name). I may take this one on myself as it is mindless and involves no technical skill but lashings of OCDness ;)

**Comment by Wolfenswan** [comment]
RegEx snippet to catch all groups:
```Grp([A-Z]{3})_([A-Z]{2,})(\d*)```

**Comment by ferstaberinde** [comment]
The blank addition might be new - it wasn't doing that when I worked on this at the weekend (hence my notes about adding the blank). Will re-test when I look at this.

**Comment by ferstaberinde** [comment]
All group IDs and callsigns added in the editor. @Wolfenswan I'm happy to close this if you are?

**Comment by ferstaberinde** [comment]
PS I removed all the '= group this' commands from inits using the magic of regex and more regex.

**Comment by Wolfenswan** [comment]
Can be closed. We can create a new issue if problems arise with the current implementation.
